File: core/src/main/java/com/emarsys/core/database/repository/AbstractSqlSpecification.java
package com.emarsys.core.database.repository;

public abstract class AbstractSqlSpecification implements SqlSpecification {
    @Override
    public boolean isDistinct() {
        return false;
    }

    @Override
    public String[] getColumns() {
        return null;
    }

    @Override
    public String getSelection() {
        return null;
    }

    @Override
    public String[] getSelectionArgs() {
        return null;
    }

    @Override
    public String getGroupBy() {
        return null;
    }

    @Override
    public String getHaving() {
        return null;
    }

    @Override
    public String getOrderBy() {
        return null;
    }

    @Override
    public String getLimit() {
        return null;
    }
}

File: core/src/main/java/com/emarsys/core/database/repository/AbstractSqlSpecification.kt
package com.emarsys.core.database.repository

abstract class AbstractSqlSpecification : SqlSpecification {
    override val isDistinct: Boolean
        get() = false
    override val columns: Array<String>?
        get() = null
    override val selection: String?
        get() = null
    override val selectionArgs: Array<String>?
        get() = null
    override val groupBy: String?
        get() = null
    override val having: String?
        get() = null
    override val orderBy: String?
        get() = null
    override val limit: String?
        get() = null
}